src/apps/filters/h3ToComponents.c

```diff
@@ -100,33 +83,64 @@ void doCell(H3Index h, int verboseMode) {
                    resDigitToChar(H3_GET_INDEX_DIGIT(h, i)));
         }
         printf("╚════════════╝\n\n");
+    } else {
+        if (h3Mode == H3_HEXAGON_MODE) {
+            printf("%d:%d:%d:", h3Mode, h3Res, h3BaseCell);
+            for (int i = 1; i <= h3Res; i++) {
+                printf("%c", resDigitToChar(H3_GET_INDEX_DIGIT(h, i)));
+            }
+            printf("\n");
+        } else if (h3Mode == H3_UNIEDGE_MODE) {
+            printf("%d:%d:%d:%d:", h3Mode, H3_GET_RESERVED_BITS(h), h3Res,
+                   h3BaseCell);
+            for (int i = 1; i <= h3Res; i++) {
+                printf("%c", resDigitToChar(H3_GET_INDEX_DIGIT(h, i)));
+            }
+            printf("\n");
+        } else {
+            printf("INVALID INDEX\n");
```

Comment:
Doesn't look like we check `h3IsValid` here, just the mode - should we check if the input is valid here? Not sure if that's necessary for all the filters, but almost certainly necessary for this one (plus this filter doesn't have perf concerns).

Reply:
This is a good point that all filters should handle invalid input, but I am going to defer this to a later change.